### PR TITLE
Fix #581 - Set map bounds using screen size in zoomToRegion()

### LINKED ISSUE #581: IllegalStateException - Map size can't be 0
**Summary:** 

From Android Developer Console for v2.0.14:

~~~
java.lang.IllegalStateException: Error using newLatLngBounds(LatLngBounds, int): Map size can't be 0. Most likely, layout has not yet occured for the map view.  Either wait until layout has occurred or use newLatLngBounds(LatLngBounds, int, int, int) which allows you to specify the map's dimensions.
	at maps.f.g.b(Unknown Source)
	at maps.E.a.a(Unknown Source)
	at maps.af.k$2.a(Unknown Source)
	at maps.E.a.a(Unknown Source)
	at maps.af.t.b(Unknown Source)
	at vl.onTransact(:com.google.android.gms.DynamiteModulesB:92)
	at android.os.Binder.transact(Binder.java:380)
	at com.google.android.gms.maps.internal.IGoogleMapDelegate$zza$zza.animateCamera(Unknown Source)
	at com.google.android.gms.maps.GoogleMap.animateCamera(Unknown Source)
	at org.onebusaway.android.map.googlemapsv2.BaseMapFragment.zoomToRegion(BaseMapFragment.java:606)
	at org.onebusaway.android.map.googlemapsv2.BaseMapFragment$MapDialogFragment$2.onClick(BaseMapFragment.java:960)
	at android.support.v7.app.AlertController$ButtonHandler.handleMessage(AlertController.java:153)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5343)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:905)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:700)
~~~

**Steps to reproduce:** 

Unknown

**Expected behavior:** 

Not crash

**Observed behavior:** 

Crash

**Device and Android version:** 

~~~
Moto X (1st Gen) (ghost)
Manufacturer Motorola 
Android version Android 4.2 
RAM (MB) 2048 
Screen size 720 × 1280 
Screen density (dpi) 320 
OpenGL ES version 2.0 
Native platform armeabi-v7a 
CPU make Qualcomm 
CPU model MSM8960
~~~

### PR DESCRIPTION
**Pull request by therajanmaurya**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Sign the [electronic OBA ICLA](https://docs.google.com/forms/d/12jV-ByyN186MuPotMvxJtNKtSaGGTnEHm8rXomM2bm4/viewform)

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `gradlew connectedObaGoogleDebugAndroidTest connectedObaAmazonDebugAndroidTest` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

- Add : Dynamic Map size, according to device screen size .

### CODE REVIEW

**Comment by therajanmaurya** [comment]
Hi @barbeau Please review.

And I did not sign the ICLA till now. Is it necessary for this PR to merge to sign the ICLA.

**Comment by barbeau** [comment]
ICLA manually verified :+1:

File: onebusaway-android/src/amazon/java/org/onebusaway/android/map/googlemapsv2/BaseMapFragment.java [hunk 1]
```diff
@@ -654,8 +654,13 @@ public void zoomToRegion() {
 
         if (region != null && mMap != null) {
             LatLngBounds b = MapHelpV2.getRegionBounds(region);
-            int padding = 0;
-            mMap.animateCamera((CameraUpdateFactory.newLatLngBounds(b, padding)));
+
+            //Setting the width and height of your screen
+            int width = getResources().getDisplayMetrics().widthPixels;
+            int height = getResources().getDisplayMetrics().heightPixels;
+            int padding = (int) (width * 0.12); // offset from edges of the map 12% of screen
```
Comment (barbeau) [inline onebusaway-android/src/amazon/java/org/onebusaway/android/map/googlemapsv2/BaseMapFragment.java]: Any reason in particular that exactly 12% is used here?

**Comment by barbeau** [comment]
@therajanmaurya Looks like you modified the file in the path `/src/amazon/java/` - you'll need to make these changes to file in `/src/google/java/` instead, and then run the app - the code for `/src/amazon/java/` (for Amazon FirePhone) will then be automatically generated.  The end result should be edits to `BaseMapFragment` in both paths.

**Comment by therajanmaurya** [comment]
@barbeau it's my mistake, Sorry for that to forget to change the build variant.
Now it's done and I updated the PR with build variant obaGoogleDebug.
I have changed the padding = 0 and documented the code, what's the use of padding = 0.
Reference for more about the padding = 0 : http://stackoverflow.com/a/27937256/3176277

**Comment by barbeau** [comment]
@therajanmaurya Are you actually installing this patch on an emulator or device?  The build script should generate the edits to `BaseMapFragment` for the Amazon build variant as well.  So you have have two files edited in this PR is all goes well.

**Comment by therajanmaurya** [comment]
@barbeau Yes, I am installing this patch to Moto G 3 turbo Edition. 
and I have checked to change the build variant to Amazon and yes, It's showing BaseMapFragment is changed as change occur in Google variant.

**Comment by barbeau** [comment]
You should just leave the build variant on  obaGoogleDebug.  Sounds like
the Amazon changes didn't get committed with the Google changes.  Can you
please update PR with single commit that includes changes to both files (as
generated by build script)?

**Comment by therajanmaurya** [comment]
you mean, make the commit again after changing the build variant to Amazon and then squash the both commit.

**Comment by barbeau** [comment]
No, don't change to Amazon variant.  Leaving it on obaGoogleDebug.  Build
the app, and then squash any changes from master into one commit for this
PR.

The only reason for changing to the Amazon variant is if you have a Fire
Phone and want to run the app on it. If you are building on a normal Google
device, don't change to the Amazon variant.

**Comment by therajanmaurya** [comment]
@barbeau it's done as you said. commit the changes of obaGoogleDebug and did not change to Amazon variant and with Squash !

**Comment by barbeau** [comment]
@therajanmaurya Somethings not going right - I'm still only seeing one file changed in this commit.  There should be two, one for each build variant.  You get this by following these steps:

1. Checkout master branch
1. Checkout new branch for this PR
1. Select `obaGoogleDebug` build variant
1. Make changes to the `BaseMapFragment` class in the `/google/` file path
1. Build and run app

At this point, in Android Studio you should see changes to `BaseMapFragment` in both the google and amazon file paths.

1. Commit these changes and force push to your branch for this PR.

**Comment by therajanmaurya** [comment]
Thanks @barbeau for the guideline. It's done now and perfects.
Actually, I was doing git commit before running on the device because When I was done first then I run on the device and on the second time not running on the device and making git commit.

**Comment by barbeau** [comment]
@therajanmaurya Thanks, looks better know.  I'll need to test, and it might be at least a few days until I get to that.

**Comment by therajanmaurya** [comment]
@barbeau It's Ok, Can you tell me more about the build script ?

**Comment by barbeau** [comment]
Sure.  For discussion off-topic from this PR/issue I'd head over to Slack,
and we can discuss in #android room there.

Sean J. Barbeau, Ph.D.
sjbarbeau@gmail.com
http://www.linkedin.com/in/seanbarbeau

**Comment by barbeau** [comment]
@cagryInside Could you please review/test this PR and see if all looks ok?

**Comment by barbeau** [comment]
Actually, looking at the crash reports, we're seeing a large number of reports for this issue (115 in the last week).  @cagryInside Could you review today?  I'd like to get a hotfix out for this.

**Comment by cagryInside** [comment]
> Actually, looking at the crash reports, we're seeing a large number of reports for this issue (115 in the last week). @cagryInside Could you review today? I'd like to get a hotfix out for this.
>  Show all checks

@barbeau Sure, I am looking at it now.

**Comment by cagryInside** [comment]
@barbeau I couldn't reproduce the issue. However, I tested this PR with different phones. All are working fine. Further, based on the stackoverflow entry that you posted (http://stackoverflow.com/a/30497039/937715) and my internet research on this issue, this is the accepted fix.

**Comment by barbeau** [comment]
Thanks @cagryInside for reviewing!  Looks good to me too.  Thanks @therajanmaurya for the contribution!

**Pull request merged**